### ISSUE TITLE
AsyncFileImpl - flush() and close() use the internal ordered task queue

### ISSUE THREAD

**Issue by erichhsun** [eclipse-vertx/vert.x#3331]
### Version
3.5.2, 3.8.5

### Context
Performing flush and close on multiple AsyncFiles results in each one being closed synchronously, one by one, since they are submitted to the internal ordered task queue. This results in performance degradation. 

Originally these operations were executed on the event loop, but were moved since they are blocking operations that can take a while. However, I don't think the intention was to execute them one by one - though there could be something I am missing.

See https://github.com/eclipse-vertx/vert.x/blob/master/src/main/java/io/vertx/core/file/impl/AsyncFileImpl.java#L434

and

https://github.com/eclipse-vertx/vert.x/blob/master/src/main/java/io/vertx/core/file/impl/AsyncFileImpl.java#L542

I think it would be more appropriate to send them to an unordered task queue / worker pool so that they can be flushed or closed in parallel.

### Do you have a reproducer?
Obtain multiple references of AsyncFile by using fileSystem().open() and call flush or close on each one.

### Steps to reproduce
See above.

**Comment by vietj** [comment]
@erichhsun ordering needs to be guaranteed in order to guarantee that all operations on the same file will be correctly ordered

**Comment by vietj** [comment]
have you experienced performance issues with this ?

**Comment by erichhsun** [comment]
@vietj my mistake, you are correct. 

I forgot that existing operations on the file need to complete in order first. I thought the AsyncFileImpl operations were executing only on a single thread, but they are indeed using all available worker threads in the internal worker pool at high enough concurrency; I misread the logging I added prior to filing this issue.

The performance is related more to the length of the operations but that is unrelated to the framework.

Closing this issue, thanks for the reply!
